fili-core/src/main/java/com/yahoo/bard/webservice/application/DimensionLoader.java

```diff
@@ -0,0 +1,95 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.application;
+
+import com.yahoo.bard.webservice.config.SystemConfig;
+import com.yahoo.bard.webservice.config.SystemConfigProvider;
+import com.yahoo.bard.webservice.data.dimension.Dimension;
+import com.yahoo.bard.webservice.data.dimension.DimensionRow;
+import com.yahoo.bard.webservice.druid.client.FailureCallback;
+import com.yahoo.bard.webservice.druid.client.HttpErrorCallback;
+
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+import javax.inject.Singleton;
+
+/**
+ * DimensionLoader sends requests to the druid search query interface to get a list of dimension
+ * values to add to the dimension cache.
+ */
+@Singleton
+public class DimensionLoader extends Loader<Boolean>
+        implements BiConsumer<Dimension, DimensionRow>, Consumer<Dimension> {
```

Comment:
Not sure if implementing consumer is useful here, it is unclear that a loader is a consumer.